pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ClientBuilder.java

```diff
@@ -18,6 +18,7 @@
  */
 package org.apache.pulsar.client.api;
 
+import io.netty.util.Timer;
```

Comment:
We cannot expose Netty classes on the client API. 

Instead, we could take a `Timer` instance from the `PulsarClientImpl` constructor and have the Pulsar proxy to pass it there.

Reply:
Additionally, `Timer` is not serializable and we need all classes in public API to be it:

```
Medium: Class org.apache.pulsar.client.impl.conf.ClientConfigurationData defines non-transient non-serializable instance field timer [org.apache.pulsar.client.impl.conf.ClientConfigurationData] In ClientConfigurationData.java SE_BAD_FIELD
```

Reply:
Good point!

> Instead, we could take a Timer instance from the PulsarClientImpl constructor and have the Pulsar proxy to pass it there.

I did not understand here, does this means the PulsarClientImpl exposes the timer so that the proxy can get the timer and close it? Or we can pass the Timer through the PulsarClientImpl constructor?

Reply:
>  Or we can pass the Timer through the PulsarClientImpl constructor?

Yes, we can add it to the constructor (or leave it null to get it created internally). It's ok to do so because `PulsarClientImpl` is not part of public API (but can be used by pulsar proxy).

Reply:
Thanks for pointing it out.
In this way, how can the user share the timer instance?

Reply:
They should construct the PulsarClientImpl, not the ClientBuilder. This is not a common use, just for some application like Pulsar proxy does.